extends CharacterBody2D

@export var gravity: float = 1000.0
@export var jump_force: float = -350.0

var vivo := true

func _physics_process(delta):
	if not vivo:
		return

	# Aplica gravidade constantemente
	velocity.y += gravity * delta

	# Quando o jogador aperta a tecla "pulo", aplica forÃ§a para cima
	if Input.is_action_just_pressed("pulo"):
		velocity.y = jump_force

	# Move o personagem verticalmente
	move_and_slide()

func morrer():
	if vivo:
		vivo = false
		velocity = Vector2.ZERO
		set_physics_process(false)
		print("ðŸ’€ GAME OVER")

		# Exibe o texto na tela
		var ui = get_parent().get_node("UI/GameOverLabel")
		ui.visible = true


func _on_sensor_colisao_body_entered(body: Node2D) -> void:
	print("ðŸ’¡ Colidiu com:", body.name)

	if body.is_in_group("obstaculo"):
		print("ðŸ’¥ Colidiu com obstÃ¡culo")
		morrer() # Replace with function body.

var pontos := 0

func _on_pontuou():
	pontos += 1
	print("ðŸŽ¯ Ponto marcado:", pontos)
	$UI/PontuacaoLabel.text = str(pontos)
